pkg/command/jobctl/query.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2021 Andy Dalton
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jobctl
+
+import (
+	"context"
+	"errors"
+	"os"
+	"strconv"
+
+	"github.com/adalton/teleport-exercise/pkg/client/jobmanager"
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+)
+
+var queryCmd = &cobra.Command{
+	Use:     "query",
+	Short:   "Query job state",
+	Long:    "Query the state of a job managed by JobManager",
+	Example: "jobctl query ba90b623-3dae-4bdd-8b96-c1ea4a999c44",
+	RunE:    query,
+}
+
+func init() {
+	rootCmd.AddCommand(queryCmd)
+}
+
+func query(cmd *cobra.Command, jobIDs []string) error {
+	if len(jobIDs) == 0 {
+		return errors.New("no jobs specified")
+	}
+
+	c, err := jobmanager.NewClient(argUserID, argServerHostPort)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+	var lastError error
+
+	jobStatusList := make([]*jobmanager.JobStatus, 0, len(jobIDs))
+
+	for _, jobID := range jobIDs {
+		func() {
+			ctx, cancel := context.WithTimeout(cmd.Context(), shortOperationTimeout)
```

Comment:
Suggestion: ditch the anonymous func and defer in the loop (seem alright here)? You could aways just cancel() after the query too.

Same for similar idioms.